php-checks/src/main/java/org/sonar/php/checks/security/CORSPolicyCheck.java

```diff
@@ -0,0 +1,115 @@
+public class CORSPolicyCheck extends PHPVisitorCheck {
+
+  private static final String MESSAGE = "Make sure this permissive CORS policy is safe here.";
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree functionCallTree) {
+    FunctionCallHelper functionCallHelper = FunctionCallHelper.create(functionCallTree);
+
+    if ((functionCallHelper.isResponseConstuctorVulnerable()) || (functionCallHelper.isCoreHeaderVulnerable()) || (functionCallHelper.isSetOrHeaderVulnerable())) {
+      context().newIssue(this, functionCallTree, MESSAGE);
+    }
+    super.visitFunctionCall(functionCallTree);
+  }
+
+  private static class FunctionCallHelper {
+
+    private static final String HEADER = "header";
+
+    private FunctionCallTree functionCallTree;
+    private ExpressionTree callee;
+
+    private FunctionCallHelper(FunctionCallTree functionCallTree) {
+      this.functionCallTree = functionCallTree;
+      this.callee = functionCallTree.callee();
+    }
+
+    static FunctionCallHelper create(FunctionCallTree functionCallTree) {
+      return new FunctionCallHelper(functionCallTree);
+    }
+
+    private boolean isResponseConstuctorVulnerable() {
+      return isResponseConstructorFunctionCall() &&
+        argument(functionCallTree, "headers", 2)
+          .map(CallArgumentTree::value)
+          .filter(ArrayInitializerBracketTree.class::isInstance)
+          .map(ArrayInitializerBracketTree.class::cast)
+          .filter(a -> a.arrayPairs()
+            .stream()
+            .anyMatch(b -> isLiteralTreeEqualsTo(b.key(), "Access-Control-Allow-Origin") && isLiteralTreeEqualsTo(b.value(), "*")))
+          .isPresent();
+    }
+
+    private boolean isCoreHeaderVulnerable() {
+      return isCoreHeaderFunctionCall() &&
+        retrieveArgumentAndVerifyItIsEqualsTo(HEADER, 0, "Access-Control-Allow-Origin:*");
+    }
+
+    private boolean isSetOrHeaderVulnerable() {
+      return isSetOrHeaderFunctionCall() &&
+        retrieveArgumentAndVerifyItIsEqualsTo("key", 0, "Access-Control-Allow-Origin") &&
+        retrieveArgumentAndVerifyItIsEqualsTo("values", 1, "*");
+    }
+
+    private boolean retrieveArgumentAndVerifyItIsEqualsTo(String name, int position, String expected) {
+      Optional<CallArgumentTree> argumentTree = argument(functionCallTree, name, position);
+      return argumentTree.isPresent() && isLiteralTreeEqualsTo(argumentTree.get().value(), expected);
+    }
+
+    private static boolean isLiteralTreeEqualsTo(ExpressionTree tree, String expected) {
+      return tree.is(REGULAR_STRING_LITERAL) && expected.equalsIgnoreCase(trimQuotes(((LiteralTree) tree).value().replaceAll("\\s", "")));
+    }
+
+    private boolean isResponseConstructorFunctionCall() {
+      return callee instanceof ClassNamespaceNameTreeImpl && "Response".equalsIgnoreCase(nameOf(callee));
+    }
+
+    private boolean isCoreHeaderFunctionCall() {
+      return callee.is(NAMESPACE_NAME) && HEADER.equals(lowerCaseFunctionName(functionCallTree));
```

Comment:
```suggestion
      return Symbols.get(functionCallTree).qualifiedName().equals(HEADER_FUNCTION_NAME);
```

Reply:
The point of this suggestion is to pass through the `FunctionSymbol ` object without verifying the Kind of tree, or to compare `qualifiedName` objects rather than `String`?

Reply:
It was more about the second one.

Reply:
I tried to implement this, but in other places where I can't get (at least I think so) the qualified name I still compare String objects. Suggestions to improve that are welcome in next review!

Reply:
When dealing with method calls, we usually want to take into account the type of the receiver. However, that's the choice we made for the purpose of this rule: it's not suprizing that we don't have good tools for that, but we can think of improving that in the future.